src/main/java/org/spongepowered/api/resource/pack/Pack.java

```diff
@@ -72,14 +72,21 @@
      */
     boolean exists(PackType type, ResourcePath path);
 
+    /**
+     * Gets all the namespaces which exist in this pack.
+     *
+     * @return The set of namespaces
+     */
+    Set<String> getNamespaces();
+
     /**
      * Gets the metadata of this pack. The {@link DataView} represented is of
      * the pack.json file in the pack root. If the pack does not contain a
-     * pack.json, {@link Optional#empty()} is returned.
+     * pack.json, it is an error.
```

Comment:
wtf, it is `pack.mcmeta`, and if that one is absent, a default one will be returned

Reply:
And what should a default mcmeta contain?

Reply:
It just contains an assumed pack version and a broken message in its description